src/main/java/org/gridsuite/directory/server/repository/DirectoryElementEntity.java

```diff
@@ -53,16 +55,17 @@ public DirectoryElementEntity update(@NonNull ElementAttributes newElementAttrib
         }
 
         if (Objects.nonNull(newElementAttributes.getAccessRights())) {
-            this.isPrivate = newElementAttributes.getAccessRights().isPrivate();
+            this.isPrivate = newElementAttributes.getAccessRights().getIsPrivate();
         }
 
         return this;
     }
 
-    public static boolean isAttributesUpdatable(@NonNull ElementAttributes newElementAttributes) {
+    public static boolean isAttributesUpdatable(@NonNull ElementAttributes newElementAttributes, ElementAttributes elementAttributes, String userId) {
```

Comment:
To simplify 
```
    public boolean isAttributesUpdatable(@NonNull ElementAttributes newElementAttributes, String userId) {
        return (// Updatable attributes
            StringUtils.isNotBlank(newElementAttributes.getElementName()) ||
                    //Only the owner can update the accessRights of a directory (to avoid user locking themselves out of a directory they don't own
                    type.equals(DIRECTORY) && Objects.nonNull(newElementAttributes.getAccessRights()) && userId.equals(owner))
            && // Non updatable attributes
            Objects.isNull(newElementAttributes.getElementUuid()) &&
            Objects.isNull(newElementAttributes.getType()) &&
            Objects.isNull(newElementAttributes.getOwner()) &&
            Objects.isNull(newElementAttributes.getSubdirectoriesCount()) &&
            Objects.isNull(newElementAttributes.getDescription());
    }
```

Reply:
done